### PR TITLE
Login

### PR DESCRIPTION
**Pull request by alatham21**

### CODE REVIEW

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -52,17 +51,8 @@ class RegistrationsController < Devise::RegistrationsController
...
-    super(resource)
-    @user = current_user
-    domain = @user.email.split('@').last
-    @role = if domain == 'tamu.edu'
-              Role.where(role_name: 'Student')
-            else
-              Role.where(role_name: 'Spartan Tutor')
-            end
-    @user.roles << @role
-    flash[:notice] = 'Assigned to Student Role.'
```
Comment (alatham21) [inline app/controllers/registrations_controller.rb:65]: This no longer worked after adding email confirmation. moved to the user model.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -131,7 +129,8 @@ def admin_view_hours
   private
 
   def user_params
-    params.require(:user).permit(:first_name, :last_name, :major, :email, :encrypted_password)
+    params.require(:user).permit(:first_name, :last_name, :major, :email, :encrypted_password,
+                                 role_ids: [])
```
Comment (alatham21) [inline app/controllers/users_controller.rb:137]: because of has many through

File: app/models/user.rb [hunk 3]
```diff
@@ -20,13 +20,32 @@ class User < ApplicationRecord
 
   validates :first_name, :last_name, :email, presence: true
   validate :email_domain
+
+  after_create :add_student_role, :add_spartan_tutor_role
```
Comment (alatham21) [inline app/models/user.rb:33]: only adds one of the two roles

File: app/views/users/admin_edit.html.erb [hunk 4]
```diff
@@ -34,8 +34,11 @@
         </div>
 
         <div class="field">
-          <%= f.label :role %><br />
-          <%= f.collection_select :id, Role.order(:role_name),:id,:role_name, {include_blank: true}, {required: true, class: "form-select"} %>
+          <% Role.all.each do |role| %>
+            <%= check_box_tag "user[role_ids][]", role.id, @user.role_ids.include?(role.id) %>
+            <%= role.role_name %>
+            <br />
+          <% end %>
```
Comment (alatham21) [inline app/views/users/admin_edit.html.erb:41]: because of has many through

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -11,7 +11,7 @@
 
   <div class="row">
     <!-- Spartan tutoring check in -->
-    <% if @user.id == current_user.id and current_user.student? and !@spartan_session.blank? %>
+    <% if @user.id == current_user.id and current_user.roles.include?(Role.student_role) and !@spartan_session.blank? %>
```
Comment (baileyjrob) [inline app/views/users/show.html.erb:14]: Nice correction for standardizing

File: db/schema.rb [hunk 6]
```diff
@@ -35,6 +35,17 @@
...
+    t.bigint "course_id", null: false
+    t.bigint "tutoring_session_id", null: false
+  end
+
+  create_table "courses_users", id: false, force: :cascade do |t|
+    t.bigint "user_id", null: false
+    t.bigint "course_id", null: false
+    t.string "grade_achieved"
+  end
+
```
Comment (baileyjrob) [inline db/schema.rb]: This and other additions are incorrect. The format is "singular"_"plural". I don't know why there are duplicates in here

**Review by baileyjrob — CHANGES_REQUESTED** [review]
A few changes are needed on the backend

File: db/schema.rb (hunk 6 above)
Comment (alatham21) [reply db/schema.rb]: I'm not sure why these remained after renaming. Does renaming just make a copy? Regardless, I dropped the plural_plural tables

File: db/schema.rb [hunk 7]
```diff
@@ -149,17 +114,5 @@
...
-  add_foreign_key "courses_users", "users"
-  add_foreign_key "departments_tutoring_sessions", "departments"
-  add_foreign_key "departments_tutoring_sessions", "tutoring_sessions"
-  add_foreign_key "roles_users", "roles"
-  add_foreign_key "roles_users", "users"
-  add_foreign_key "spartan_sessions_users", "spartan_sessions"
-  add_foreign_key "spartan_sessions_users", "users"
   add_foreign_key "tutoring_sessions", "users", column: "tutor_id"
-  add_foreign_key "tutoring_sessions_users", "tutoring_sessions"
-  add_foreign_key "tutoring_sessions_users", "users"
```
Comment (baileyjrob) [inline db/schema.rb]: Missing foreign-key restraints for the proper join tables. I'm not entirely sure why this is happening, the current dev branch has the proper tables. Are you sure your local dev branch is up-to-date?

**Review by baileyjrob — CHANGES_REQUESTED** [review]
Something's still up on the back-end

File: db/schema.rb (hunk 7 above)
Comment (baileyjrob) [reply db/schema.rb]: For some reason, migrating is causing the old tables to reappear. I'm still looking into this, as none of the migrates should do this.

Comment (alatham21) [reply db/schema.rb]: so strange, but I think it is good now

**Comment by alatham21** [comment]
Checking on back end shenanigans

**Pull request closed without merging**